test/SILOptimizer/definite_init_closures.swift

```diff
@@ -0,0 +1,50 @@
+// RUN: %target-swift-frontend -emit-sil %s -o /dev/null
+
+// Test boolean operators with implicit closures
+
+struct Simple {
+  let x: Bool
+  let y: Bool
+
+  init() {
+    x = false
+    y = false || x
+  }
+}
+
+struct NestedClosures {
+  let x: Bool
+  let y: Bool
+  let z: Bool
+
+  init(_ a: Bool) {
+    x = false
+    y = false
+    z = false || (y || (x || a))
```

Comment:
Does your code behave the same with explicit "self." qualification too?

Reply:
yes. I added a test